### PR TITLE
Add `cookiesJsonFile` option

### PR DESCRIPTION
**Pull request by kyanny**
Hi @garris 
This is pull request for https://github.com/garris/BackstopJS/issues/44.

Add patch for capture/echoFiles.js and capture/genBitmaps.js.
If `scenario` has `cookiesJsonFile` property, BackstopJS assumes it as a path of JSON file, reads cookies from the file, and sets cookies to PhantomJS object.
This enables BackstopJS to take screen capture for login-required pages.

Add explanation of `cookiesJsonFile` property to README.md.
It includes short explanation and example contents of `cookiesJsonFile`.
It helps users to learn what it is, and how to deal with it.

### CODE REVIEW

File: capture/genBitmaps.js [hunk 1]
```diff
@@ -69,6 +69,13 @@ function capturePageSelectors(url,scenarios,viewports,bitmaps_reference,bitmaps_
 
 	casper.each(scenarios,function(casper, scenario, scenario_index){
 
+		if (fs.isFile(scenario.cookiesJsonFile)) {
```
Comment (garris) [inline capture/genBitmaps.js]: Please check for property value here eg., if(scenario.cookiesJsonFile && fs.isFile(...

**Comment by garris** [comment]
@kyanny, looks good.  One more thing.  Please add a way for us to verify that our cookie values are working.  See /test/simple.html line 76 -- we need to add a similar display and $scope property for the cookie string here.  Thanks!

**Comment by garris** [comment]
Sorry this is taking so long.   Will absolutely try to get this in this weekend!   Thanks for your patience.

**Comment by kyanny** [comment]
Hi @garris sorry taking long time.
I've pushed property check code.

I'm not good at angularjs, and I'm not sure how to run test and assertion.
Does this diff look good to you?

```diff
diff --git a/test/backstop.json b/test/backstop.json
index 614ed86..ddce34b 100644
--- a/test/backstop.json
+++ b/test/backstop.json
@@ -29,6 +29,20 @@
 			],
 			"readyEvent":"backstop.ready",
 			"delay":0
+		},
+		{
+			"testName":"simple",
+			"url":"test/simple.html",
+			"hideSelectors": [
+			],
+			"removeSelectors": [
+			],
+			"selectors":[
+				"body"
+			],
+			"readyEvent":"backstop.ready",
+		  "delay":0,
+                  "cookiesJsonFile":"cookies.json"
 		}
 	]
 }
diff --git a/test/cookies.json b/test/cookies.json
new file mode 100644
index 0000000..4a5e26e
--- /dev/null
+++ b/test/cookies.json
@@ -0,0 +1,5 @@
+{
+  "name": "mycookie",
+  "value": "1",
+  "path": "/"
+}
diff --git a/test/simple.html b/test/simple.html
index 2b8672f..a755bd7 100644
--- a/test/simple.html
+++ b/test/simple.html
@@ -73,7 +73,8 @@
 	    <div class="row">
 	    	
 	    	<div class="col-sm-12">
-	    		<p><strong>User agent:</strong> {{userAgentStr}}</p>
+	    	  <p><strong>User agent:</strong> {{userAgentStr}}</p>
+		  <p><strong>Cookie:</strong> {{cookieStr}}</p>
 	    	</div>
 
 	    </div>
@@ -96,7 +97,8 @@
 		}
 
 		$scope.userAgentStr = navigator.userAgent;
-
+		$scope.cookieStr = document.cookie;
+	  
 	}]);
 
 
```

**Comment by garris** [comment]
@kyanny, this looks great!   You did it perfectly.   No need for assertion -- this is simply a visual test.   I will try to test this myself hopefully tomorrow.   Thank you very much for your effort!

**Comment by kyanny** [comment]
@garris You're welcome! Thank you! :smile:

**Comment by garris** [comment]
@kyanny   Almost there -- I found some issues when I tested the patch.  Please see issue #44 for an update.  Thank you.

**Comment by kyanny** [comment]
@garris oh I will have a look #44 later. sorry my bad!

**Comment by filipecatraia** [comment]
@kyanny This is brilliant!

**Pull request merged**